File: problem387.cs
public class Solution {
    public int FirstUniqChar(string s) {
        int[] counts = new int[26];
        foreach (char letter in s) {
            counts[letter - 'a']++;
        }

        for (int i = 0; i < s.Length; i++) {
            if (counts[s[i] - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: su19/problem387.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        counts = {}
        for char in s:
            if char not in counts:
                counts[char] = 0
            counts[char] += 1
        i = 0
        while i < len(s):
            if counts[s[i]] == 1:
                return i
            i += 1
        return -1